packages/manager/src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,305 @@
+            >
+              {option.label}
+            </Box>
+            <SelectedIcon visible={state.selected} />
+          </>
+        </ListItem>
+      );
+    },
+    [renderOption]
+  );
+
+  React.useEffect(() => {
+    /**
+     * We want to clear the input value when the value prop changes to null.
+     * This is for use cases where a user changes their region and the Linode
+     * they had selected is no longer available.
+     */
+    if (defaultValue === null) {
+      setInputValue('');
+    }
+  }, [defaultValue]);
+
+  return (
+    <MuiAutocomplete
+      {...rest}
+      getOptionLabel={(option: OptionType) =>
+        renderOptionLabel ? renderOptionLabel(option) : option.label
+      }
+      isOptionEqualToValue={(option: OptionType, value: OptionType) => {
+        return option.value === value.value;
+      }}
+      noOptionsText={
+        noOptionsText || (
+          <i>
+            {getDefaultNoOptionsMessage({
+              errorText,
+              options,
+            })}
+          </i>
+        )
+      }
+      onChange={(_, selectedOptions: OptionType | OptionType[], reason) => {
+        const initialOptions = [...options]; // This is to bypass readonly options
+        handleChange({
+          handleMultiSelectionChange,
+          initialOptions,
+          onSelectionChange,
+          reason,
+          selectedOptions,
+        });
+      }}
+      renderInput={(params) => (
+        <TextField
+          errorText={errorText}
+          helperText={helperText}
+          inputId={params.id}
+          label={label}
+          loading={loading}
+          noMarginTop={noMarginTop}
+          placeholder={placeholder || 'Select an option'}
+          {...params}
+        />
+      )}
+      ChipProps={{ deleteIcon: <CloseIcon /> }}
+      PopperComponent={CustomPopper}
+      clearOnBlur={clearOnBlur}
+      defaultValue={defaultValue}
+      disableCloseOnSelect={multiple}
+      disablePortal={disablePortal}
+      inputValue={inputValue}
+      limitTags={limitTags}
+      loading={loading}
+      loadingText={loadingText || 'Loading...'}
+      multiple={multiple}
+      onBlur={onBlur}
+      onInputChange={(_, value) => setInputValue(value)}
+      options={multiple ? optionsWithSelectAll : options}
+      popupIcon={<KeyboardArrowDownIcon />}
+      renderOption={handleRenderOption}
+      sx={sx}
+      value={multiple ? selectedOptions : undefined} // Only pass value prop for multi-select mode
+    />
+  );
+};
+
+const enum NoOptionsMessage {
+  Error = 'An error occurred while fetching your options',
+  NoOptions = 'You have no options to choose from',
+  NoResults = 'No results',
+}
+
+const getDefaultNoOptionsMessage = ({
+  errorText,
+  options,
+}: DefaultNoOptionsMessage): NoOptionsMessage => {
+  if (errorText) {
+    return NoOptionsMessage.Error;
+  } else if (!options?.length) {
+    return NoOptionsMessage.NoOptions;
+  } else {
+    return NoOptionsMessage.NoResults;
+  }
```

Comment:
We can unnest the ifs since each option returns out of the function
```suggestion
  if (errorText) {
    return NoOptionsMessage.Error;
  }
  if (!options?.length) {
    return NoOptionsMessage.NoOptions;
  }
  return NoOptionsMessage.NoResults;
```

Reply:
I have no preference since it's interchangeable - open to feedback from others on both conventions

Reply:
I would say the second one looks a little more readable.